### ISSUE TITLE
#stop_stream seems not stop auto reconnect

### ISSUE THREAD

**Issue by sinorga** [tweetstream/tweetstream#200]
gem version: 2.6.1
After received `on_reconnect` callback, I call the `#stop_stream` to stop the stream.
but it still keeps receiving the `on_reconnect` callback until reached to max retries.
Is it expected behavior?

**Comment by sinorga** [comment]
I found the problem is when reconnecting succeed, it reset the connection parameters in `em-twitter` gem. so if you want to stop the stream, it should be called in error callbacks.

**Comment by stve** [comment]
Hi @sinorga 

That's correct. The `on_reconnect` informs you of a reconnect (while also reconnecting you). What did your code look like? Just curious to see if this is something that could be handled differently. Thanks.

**Comment by sinorga** [comment]
Hi @stve 

Sorry, I didn't notice this comment until now. 
At the beginning, I got a lot of reconnecting events and I didn't know what the exact error is. I need to stop it from keeping reconnecting to twitter in such situation. The `on_error` callback seems not designed for the common case, so I tried to stop it in `on_reconnect` callback. As you know it didn't work. I noticed when it gets into the reconnecting process, `#stop_stream` will be reset, it only works in some callbacks which will be executed before the reconnecting process. At the end I setup all callbacks to handle each situation.  😄 

I also noticed there are some callbacks (ex. `forbidden_callback`) which are exposed by `em-twitter` but not `tweetstream`. any special reasons? Maybe `tweetstream` could bypass them to `on_error` callback. :)
